arrow-fx/src/test/kotlin/arrow/fx/SemaphoreTest.kt

```diff
@@ -34,7 +34,7 @@ class SemaphoreTest : UnitSpec() {
       "$label - tryAcquire with available permits" {
         val n = 20
         semaphore(20).flatMap { s ->
-          (0 until n).toList().traverse(IO.applicative()) { s.acquire() }.flatMap {
+          (0 until n).toList().traverse(IO.applicative()) { IO.unit }.flatMap {
```

Comment:
This test should probably be simplified. No reason to work with `20` permits and `traverse` just to test `tryAqcuire` with permits left. Ideally, we should use a `NonZeroPosInt` property test here instead and pass it to `semaphore(..)`.

Reply:
This seems to fail, maybe we should just revisit them later on and improve

OutOfMemoryError: Java heap space